src/mps/projmposum.jl

```diff
@@ -128,10 +103,56 @@ ProjMPOSum `P`, and `ortho` is a String which can take
 the values `"left"` or `"right"` depending on the
 sweeping direction of the DMRG calculation.
 """
-function noiseterm(P::ProjMPOSum, phi::ITensor, dir::String)
+function noiseterm(P::ProjSum, phi::ITensor, dir::String)
   nt = noiseterm(P.pm[1], phi, dir)
   for n in 2:length(P.pm)
     nt += noiseterm(P.pm[n], phi, dir)
   end
   return nt
 end
+
+"""
+A ProjMPOSum computes and stores the projection of an
+implied sum of MPOs into a basis defined by an MPS,
+leaving a certain number of site indices of each MPO
+unprojected. Which sites are unprojected can be shifted
+by calling the `position!` method. The MPOs used as
+input to a ProjMPOSum are *not* added together beforehand;
+instead when the `product` method of a ProjMPOSum is invoked,
+each projected MPO in the set of MPOs is multiplied by
+the input tensor one-by-one in an efficient way.
+
+Drawing of the network represented by a ProjMPOSum
+`P([H1,H2,...])`, showing the case of `nsite(P)==2`
+and `position!(P,psi,4)` for an MPS `psi` (note the
+sum Σⱼ on the left):
+
+```
+     o--o--o-      -o--o--o--o--o--o <psi|
+     |  |  |  |  |  |  |  |  |  |  |
+ Σⱼ  o--o--o--o--o--o--o--o--o--o--o Hⱼ
+     |  |  |  |  |  |  |  |  |  |  |
+     o--o--o-      -o--o--o--o--o--o |psi>
+```
+"""
+const ProjMPOSum = ProjSum{ProjMPO}
+
+"""
+A `DiskProjMPOSum` functions the same as a `ProjMPOSum`
+(see the docstring for `ProjMPOSum`) but automatically caches most of
+the tensors it stores onto the hard drive to save RAM memory.
+"""
+const DiskProjMPOSum = ProjSum{DiskProjMPO}
```

Comment:
Given all the functions are/will be generic in terms of `ProjSum`, I don't think we even need to define this type. It shouldn't be something any user needs anyway since they can just use the `disk` function.